.gitlab/dev_container_deploy/e2e.yml

```diff
@@ -21,9 +21,7 @@ qa_agent_ot:
   extends: .docker_publish_job_definition
   stage: dev_container_deploy
   rules:
-    # incident-28463: this job is currently broken
-    # - !reference [.on_container_or_e2e_changes]
-    - !reference [.manual]
+    - !reference [.on_container_or_e2e_changes]
```

Comment:
This rule does not really make sense because `containers` test does not need the OT agent image. Would it be an issue for you to keep the job manual for now? Until we have some tests running in the CI that need this image

Reply:
I think @liustanley is going to be adding these soon, so I was just jumping ahead. I don't really think this causes much harm as-is, but I am fine with keeping it manual otherwise. I just wanted to make Stanley's life easier by making sure the containers were available for him as the tests get developed.

Reply:
Yep that's not really an issue to have this rule. From what I saw on the @liustanley the rules that we'll be used is likely to be `.on_otel_or_e2e_changes` (but you will not be able to put it in your job before the PR adding otel e2e tests is merged). But that's definitely not a blocker and feel free to merge it as is